### DISCUSSION TITLE
Timeline for LazyGraphRAG Integration into GraphRAG Library

### DISCUSSION THREAD

**Discussion by KentonParton** [microsoft/graphrag#1490] [General]
[This announcement](https://www.microsoft.com/en-us/research/blog/lazygraphrag-setting-a-new-standard-for-quality-and-cost/) mentioned that LazyGraphRAG will be coming soon to the GraphRAG library. Can you share an estimated timeline or any updates on when we can expect this feature to be added?

Thanks!

**Comment by AlonsoGuevara** [comment]
Hi @KentonParton
Lazy GraphRAG is indeed the next milestone for our repo. We are just closing some last items like clean up and ergonomics for our current implementation and Lazy GraphRAG will be the next top priority item to release.

Thanks for youe interest in this.

**Reply by KentonParton** [reply]
Thanks for the info @AlonsoGuevara. Do you have a rough estimate of when it will be complete?

Like this week, next week, this month, January, February etc?

**Reply by davidliudev** [reply]
Thanks for the great work for this great library. @AlonsoGuevara  . Our team is very interested in LazyGraphrag and is also wondering about the timeline for a release. Do you have any estimates on this?

**Reply by ziudeso** [reply]
Same with our team, we'd love know even a rough estimate, thanks for the amazing job

**Reply by puneetpawaia** [reply]
Hi. We are also interested in this implementation and would like to know if there is an estimate on the time of availability.

**Reply by MingjunHu** [reply]
我也时 对此非常按兴趣，很期待。

**Reply by ssupinma** [reply]
We are waiting for the release.

**Reply by triple888** [reply]
+1 interest on this

**Reply by zhangjiekui** [reply]
+1

**Reply by owquresh** [reply]
+1

**Reply by Bugbait** [reply]
Really looking forward to this repo! I hope it's still an active project!

**Reply by AxelSjoberg** [reply]
Thanks for the great work on this library, our team is also super excited about the release

**Reply by fredrikramberg** [reply]
I'm very excited about the release of LazyGraphRAG! My friend and I are even planning write our master's thesis on it. Do you have any updates on the expected release date? It would really help us plan our work. Thanks in advance!

**Reply by oscarhoffmann3487** [reply]
I'm really enthusiastic about this release as well!

**Reply by marcusze** [reply]
+1 interest on this

**Reply by alsulaimi88** [reply]
Looking forward to this +1

**Reply by Bugbait** [reply]
Hmm. Getting a bad feeling on the delays. Is this project dead?

**Reply by adrianzartis** [reply]
+1

**Reply by taprosoft** [reply]
+1 spent a long time waiting

**Reply by sunwei3126** [reply]
+1 waiting for testing

**Reply by fackweb** [reply]
+1 waiting for testing

**Reply by tushengtao** [reply]
+1 waiting for testing

**Reply by owquresh** [reply]
Is this project still being maintained? I would not be surprised if Microsoft has shelved it.

**Reply by jetdream** [reply]
I start questioning if it is even real. The single article and nothing else - no even initial implementation code in repo, no mentions on conferences and other public events.
But the idea itself sounds legit.
It is also weird that other GraphRAG implementations (llamaindex, langgraph) did not try to do anything like this on their own.

**Reply by bernaljg** [reply]
Perhaps you guys can try HippoRAG 2 instead! It outperforms GraphRAG in several benchmarks and is around 10x more efficient. https://github.com/OSU-NLP-Group/HippoRAG

**Reply by ntsarb** [reply]
+1 waiting for testing and deployment.

**Reply by jochenvw** [reply]
Also waiting :)

**Comment by Sandy4321** [comment]
pls release lazygraphrag...

**Comment by BrainFartScientist** [comment]
Just curious: When will it be released?

**Reply by sivarudramurthy** [reply]
Is this shelved?

**Comment by Govind-S-B** [comment]
Just commenting here to say folks are still looking for some update

**Comment by nisarg-refyne** [comment]
Hi @AlonsoGuevara,
Is there any estimated timeline around when it would be published?

**Comment by jeremyzhangsq** [comment]
Hello everyone,

We've been working on KET-RAG, a framework built upon Microsoft's GraphRAG, aiming to enhance indexing efficiency—a goal similar to LazyGraphRAG. While awaiting the official release of LazyGraphRAG, we wanted to share our approach and findings, hoping they might contribute to the community's exploration of efficient indexing strategies.

[Technical Report](https://arxiv.org/abs/2502.09304) 
[Code & Implementation](https://github.com/waetr/KET-RAG)

We welcome any feedback or thoughts on our approach.

**Reply by btebbutt** [reply]
have you checked out fast graph rag repo (circle ci) which I believe uses hippo rag. on first glance this ket rag looks quite similar and i'm interested as to whether or not this method builds upon hippo rag / is better in some scenarios etc.

**Reply by waetr** [reply]
Thank you for your comments. KET-RAG is _not_ built upon HippoRAG but MS GraphRAG; In fact, we conducted further experiments, where KET-RAG has shown superior performance on datasets/benchmarks such as Musique and RAG-QA Arena, a recent benchmark tailored for RAG tasks. We will update the experimental results in the technical report / codes on github later.

**Comment by WingI9B1** [comment]
Is this project still being maintained?

**Reply by medeirosjoaquim** [reply]
it will take a couple more years to be implemented, because it's 'lazy'
